### DISCUSSION TITLE
Creating pull request and adding assignee via API triggering two notification emails

### DISCUSSION THREAD

**Discussion by atnak** [community/community#27585] [Apps, API and Webhooks]
Partly due to the fact it's [not possible to create a PR and add assignees in a single request](https://github.com/orgs/community/discussions/24724), we're calling `github.rest.pulls.create()` followed by `github.rest.issues.addAssignees()`.  (These are `actions/github-script` methods but I presume they're just calling the respective API endpoints under the hood.)

This causes weirdness in that two notification emails are sent to the assignee instead of a more ideal and less annoying single email.

First email is an email describing the newly created PR with the final line reading "You are receiving this because you were assigned."

Second email has the body `Assigned #1234 to @assignee.`

The first email is strange in that the event itself occured before the assignee was actually assigned, but is received anyway.  This suggests there's some kind of a race if the create and update calls are performed quickly enough apart.  (We tried adding a sleep between the two calls and found that only the second email was received if we do so.)

I guess the discussion comes down to this:  The inability to add assignees to a PR during creation seems to be an oversight in design when considering email notifications, and it would be great if there was solution or workaround for this.

**Comment by atnak** [comment]
I get two email even when I create PRs / Issues from the web UI and assign during initial creation.

So in a way, it's "by design" ?

As a workaround, using a mention in the body of the PR instead of assigning them is a way of triggering only one email while still getting the assignee "subscribed" to the PR.

**Comment by jfullstackdev** [comment]
> So in a way, it's "by design" ?

Every pull request is an issue, but not every issue is a pull request. For this reason, "shared" actions for both features, like manipulating assignees, labels and milestones, are provided within [the Issues API](https://docs.github.com/en/rest/reference/issues).

https://docs.github.com/en/rest/pulls/pulls#about-the-pulls-api
